llm_studio/app_utils/hugging_face_utils.py

```diff
@@ -122,8 +122,19 @@ def publish_model_to_hugging_face(
         repo_id=repo_id, repo_type="model", commit_message="Upload model card"
     )
 
-    # push config to hub
     api = huggingface_hub.HfApi()
+
+    # push classification head to hub
+    if cfg.type == "causal_classification":
```

Comment:
You could use the already implemented `cfg.problem_type` (that maps to the filename of the config, e.g. text_causal_language_modeling).